src/test/java/com/checkmarx/flow/cucumber/integration/cli/ast/AstCliSteps.java

```diff
@@ -0,0 +1,237 @@
+
+        log.info("Jira project key: {}", JIRA_PROJECT);
+        jiraProperties.setProject(JIRA_PROJECT);
+        initJiraBugTracker();
+        log.info("reset sca filters");
+        scaProperties.setFilterSeverity(Collections.emptyList());
+
+        astProperties.setPreset("Checkmarx Default");
+        astProperties.setIncremental("false");
+    }
+
+    @After
+    public void afterEachScenario() throws IOException {
+        log.info("Cleaning JIRA project: {}", jiraProperties.getProject());
+        jiraUtils.cleanProject(jiraProperties.getProject());
+
+        if (directoryToScan != null) {
+            log.info("Deleting temp directory: {}", directoryToScan);
+            PathUtils.deleteDirectory(directoryToScan);
+        }
+    }
+
+    @Given("scanner is {}")
+    public void setScanInitiator(String initiatorList) {
+        String[] intiators ;
+
+        if(!initiatorList.contains(SEPARATOR)){
+            intiators = new String[1];
+            intiators[0] = initiatorList;
+        }
+        else{
+            intiators = initiatorList.split(SEPARATOR);
+        }
+
+        List<String> scanners = new LinkedList<>();
+        
+        for (String scanType: intiators) {
+            if(scanType.equalsIgnoreCase(ScaProperties.CONFIG_PREFIX)){
+                scanners.add(ScaProperties.CONFIG_PREFIX);
+            }
+            if(scanType.equalsIgnoreCase(AstProperties.CONFIG_PREFIX)){
+                scanners.add(AstProperties.CONFIG_PREFIX);
+            }
+
+        }
+        flowProperties.setEnabledVulnerabilityScanners(scanners);
+
+    }
+    @And("source directory contains vulnerable files")
+    public void sourceDirectoryContainsVulnerableFiles() throws IOException {
+        directoryToScan = getTempDir();
+        copyTestProjectTo(directoryToScan);
+    }
+
+
+    @When("running a AST scan with break-build on {}")
+    public void runningWithBreakBuild(String issueType) {
+        StringBuilder commandBuilder = new StringBuilder();
+        setScanInitiator("AST");
+        
+        switch (issueType) {
+            case "success":
+                commandBuilder.append("--scan --app=MyApp --cx-project=test").append(GITHUB_REPO_ARGS);
+                break;
+            case "missing-mandatory-parameter":
+                commandBuilder.append(GITHUB_REPO_ARGS);
+                break;
+            case "missing-project":
+                commandBuilder.append("--scan --app=MyApp --f=nofile").append(REPO_ARGS);
+                break;
+            default:
+                throw new PendingException("Issues type " + issueType + " isn't supported");
+        }
+
+        log.info("Running CxFlow scan with command line: {}", commandBuilder.toString());
+        Throwable exception = null;
+        try {
+            TestUtils.runCxFlow(cxFlowRunner, commandBuilder.toString());
+        } catch (Throwable e) {
+            exception = e;
+        }
+        cxFlowExecutionException = exception;
```

Comment:
why do we need the "exception" variable?

Reply:
because there's another step which is checking the exception class. In validateExitCode()

Reply:
cxFlowExecutionException  is all you need, the temporary "exception" lives for just 6 lines of code.
just use the cxFlowExecutionException  in its place

Reply:
fixed